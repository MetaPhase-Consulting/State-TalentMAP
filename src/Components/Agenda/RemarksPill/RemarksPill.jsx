import PropTypes from 'prop-types';
import FA from 'react-fontawesome';
import { EMPTY_FUNCTION } from '../../../Constants/PropTypes';

const RemarksPill = props => {
  // TODO - make use of colors?
  const { /* color, */ remark, isEditable, updateSelection } = props;


  return (
    <div className="remarks-pill" style={{ backgroundColor: '#0071bc' }}>
      {remark.text}
      { isEditable &&
        <FA name="times" onClick={() => updateSelection(remark)} />
      }
    </div>
  );
};

RemarksPill.propTypes = {
  remark: PropTypes.string,
  // color: PropTypes.string,
  isEditable: PropTypes.bool,
  updateSelection: PropTypes.func,
};

RemarksPill.defaultProps = {
  remark: '',
  // color: '#513C2C',
  isEditable: false,
  updateSelection: EMPTY_FUNCTION,
};

export default RemarksPill;
