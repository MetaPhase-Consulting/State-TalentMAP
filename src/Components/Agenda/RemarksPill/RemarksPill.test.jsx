import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import RemarksPill from './RemarksPill';

describe('RemarksPill', () => {
  const props = {
    remark: {
      active_ind: 'Y',
      order_num: 7,
      rc_code: 'B',
      seq_num: 2,
      short_desc_text: 'Promo Bd Recognized',
      text: 'Potential recognized by {Bureau} at last promo board {date}',
      remark_inserts: [
        {
          riinsertiontext: '{Bureau}',
          rirmrkseqnum: 225,
          riseqnum: 193,
        },
        {
          riinsertiontext: '{date}',
          rirmrkseqnum: 225,
          riseqnum: 192,
        },
      ],
      ari_insertions: {
        193: 'GTM/EX/OTS/SSD/OAB/OS',
        192: '1.1.1600',
      },
    },
    isEditable: true,
    userSelection: () => {},
  };

  it('is defined', () => {
    const wrapper = shallow(<RemarksPill />);
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<RemarksPill />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  xit('matches snapshot when remarks are provided', () => {
    const wrapper = shallow(<RemarksPill {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  xit('displays an editable remark', () => {
    const wrapper = shallow(<RemarksPill {...props} />);
    const expected = 'Potential recognized by GTM/EX/OTS/SSD/OAB/OS at last promo board 1.1.1600<FontAwesome />';
    expect(wrapper.find('div').text()).toBe(expected);
  });

  xit('displays an un-editable remark', () => {
    const wrapper = shallow(<RemarksPill {...props} isEditable={false} />);
    const expected = 'Potential recognized by GTM/EX/OTS/SSD/OAB/OS at last promo board 1.1.1600';
    expect(wrapper.find('div').text()).toBe(expected);
  });
});
