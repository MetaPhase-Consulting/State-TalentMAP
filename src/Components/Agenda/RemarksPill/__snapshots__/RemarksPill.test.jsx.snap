// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemarksPill matches snapshot 1`] = `
<div
  className="remarks-pill remark-category--undefined"
/>
`;

exports[`RemarksPill matches snapshot when remarks are provided 1`] = `
<div
  className="remarks-pill remark-category--B"
>
  Potential recognized by last promo board
  <FontAwesome
    name="times"
    onClick={[Function]}
  />
</div>
`;
